components/eamxx/src/share/util/scream_vertical_interpolation.hpp

```diff
@@ -128,16 +119,62 @@ void perform_vertical_interpolation_impl_1d(
   const MemberType& team,
   const LIV<T,P>& vert_interp);
 
+/* ---------------------------------------------------------------------- 
+ * Versions where x_tgt is a 2-D view
+ * ---------------------------------------------------------------------- */
 template<typename T, int P, int N> 
-void perform_vertical_interpolation_impl_Nd(
-  const view_Nd<const Pack<T,P>,2>& x_src,
+void perform_checks(
+  const view_2d<const Pack<T,P>>&   x_src,
+  const view_2d<const Pack<T,P>>&   x_tgt,
+  const view_Nd<const Pack<T,P>,N>& input,
+  const view_Nd<      Pack<T,P>,N>& output,
+  const int                         nlevs_src,
+  const int                         nlevs_tgt);
+
+template<typename T, int P> 
+void apply_interpolation(
+  const                        int  num_levs,
+  const                          T  mask_val,
+  const                   LIV<T,P>& vert_interp,
+  const view_2d<const Pack<T,P>>&   x_src,
+  const view_2d<const Pack<T,P>>&   x_tgt,
+  const view_Nd<const Pack<T,P>,2>& input,
+  const view_Nd<      Pack<T,P>,2>& output,
+  const view_Nd<        Mask<P>,2>& mask);
+
+template<typename T, int P> 
+void apply_interpolation(
+  const                        int  num_levs,
+  const                          T  mask_val,
+  const                   LIV<T,P>& vert_interp,
+  const view_2d<const Pack<T,P>>&   x_src,
+  const view_2d<const Pack<T,P>>&   x_tgt,
+  const view_Nd<const Pack<T,P>,3>& input,
+  const view_Nd<      Pack<T,P>,3>& output,
+  const view_Nd<        Mask<P>,3>& mask);
+
+template<typename T, int P> 
+void apply_interpolation(
+  const                        int  num_levs,
+  const                          T  mask_val,
+  const                   LIV<T,P>& vert_interp,
+  const view_2d<const Pack<T,P>>&   x_src,
+  const view_2d<const Pack<T,P>>&   x_tgt,
+  const view_Nd<const Pack<T,P>,4>& input,
```

Comment:
Do we really have a use case for 4d?

Reply:
Not that I can think of offhand - but it was easy to add.  In IO we support up to rank 6, so I was tempted to go that high but I think case 5 and 6 are even more unlikely.  I assume if we ever hit that someone can add the equivalent vertical interpolation case.

So TLDR; I went up to 4 because IO supports up to 6, but I thought 5 and 6 were pretty unlikely.

If you want I can remove this case.

Reply:
IO will _really_ need only up to 5, for dyn grid stuff (tracers and horiz_winds). 6 is never really used other than for dyn internal fields with time levels (but the outside world never sees the time level index).

Since dyn output is usually needed only for debug or restart purposes, I don't think we need to support rank>3. Each of these fcns need to be tested, and tests must be maintained. So if we realistically don't use it, I would just get rid of 4d.

Reply:
okay cool, I'll remove it.